echopop/utils/batch_load.py

```diff
@@ -0,0 +1,608 @@
+    )
+    # ---- Convert to lower case
+    cells_df["region_class"] = cells_df["region_class"].str.lower()
+
+    # Get the transect spacing
+    updated_intervals = export_transect_spacing(intervals_df, default_transect_spacing)
+
+    # Merge the datasets together
+    return group_merge(cells_df, [updated_intervals, layers_df]), updated_intervals
+
+
+def load_export_regions(region_files: dict, region_names: dict, root_dir: str):
+    """
+    Load region definitions.
+    """
+
+    # initialize region DataFrame
+    region_df = pd.DataFrame()
+
+    # Get filenames and read
+    for key, value in region_files.items():
+        # ---- Get region-specific filename
+        filename = value["filename"]
+        # ---- Generate the full filepath
+        full_path = Path(root_dir) / filename
+        # ---- Validate existence
+        if full_path.exists():
+            # Read file
+            new_region_df = pd.read_excel(full_path, sheet_name=value["sheetname"])
+        else:
+            raise FileNotFoundError(f"Region definition file ({str(full_path)}) not found!")
+        # ---- Fix strings if needed
+        new_region_df = new_region_df.replace('"', "", regex=True).replace(
+            r"^\s*(.*?)\s*$", r"\1", regex=True
+        )
+        # ---- Reference the `REGION_EXPORT_MAP`
+        region_settings = REGION_EXPORT_MAP[key]
+        # ---- Update column names if needed
+        new_column_names = {col: info["name"] for col, info in region_settings.items()}
+        new_region_df = new_region_df.rename(columns=new_column_names)
+        # ---- Validate column dtypes and existence
+        expected_columns = list(new_column_names.values())
+        missing_columns = [col for col in expected_columns if col not in new_region_df.columns]
+        if missing_columns:
+            raise ValueError(f"Missing columns in DataFrame for group '{key}': {missing_columns}")
+        # ---- Drop NaN values
+        new_region_df = new_region_df.dropna(subset=["transect_num", "haul_num", "region_class"])
+        # ---- Assign datatypes
+        new_region_df = new_region_df.astype(
+            {info["name"]: info["type"] for info in region_settings.values()}
+        )
+        # --- Add group ID key
+        new_region_df["group"] = key
+        # ---- Convert region class column to lower case
+        new_region_df["region_class"] = new_region_df["region_class"].str.lower()
+        # ---- Create key-specific filter pattern
+        pattern = "|".join(
+            [r"(?<!-)\b{}\b".format(re.escape(name.lower())) for name in region_names[key]]
+        )
+        # ---- Apply filter to regions dataframe
+        new_region_filtered = new_region_df[
+            new_region_df["region_class"].str.contains(pattern, case=False, regex=True)
+        ]
+        # ---- Concatenate to the initialized/pre-existing DataFrame
+        region_df = pd.concat([region_df, new_region_filtered], ignore_index=True)
```

Comment:
I think it is more memory-efficient/faster if you accumulate a list and then use `pd.concat(that_list)` to turn that into a dataframe at once.